From ed50f3bb3ce492bd6190d2c264ba00766346daf6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=B6rg=20M=C3=BCller?= <nexyon@gmail.com>
Date: Fri, 10 May 2019 22:40:59 +0200
Subject: [PATCH 13/16] Bugfix: Return correct length for modulator and
 superpose.

Also:
- Remove unnecessary includes in DynamicMusic.cpp
- Formatting fix in Sine.h
---
 include/generator/Sine.h         | 3 +--
 src/fx/DynamicMusic.cpp          | 3 ---
 src/fx/ModulatorReader.cpp       | 2 +-
 src/sequence/SuperposeReader.cpp | 2 +-
 4 files changed, 3 insertions(+), 7 deletions(-)

diff --git a/include/generator/Sine.h b/include/generator/Sine.h
index ee78fc7..0c7dd15 100644
--- a/include/generator/Sine.h
+++ b/include/generator/Sine.h
@@ -53,8 +53,7 @@ public:
 	 * \param frequency The desired frequency.
 	 * \param sampleRate The target sample rate for playback.
 	 */
-	Sine(float frequency,
-					 SampleRate sampleRate = RATE_48000);
+	Sine(float frequency, SampleRate sampleRate = RATE_48000);
 
 	/**
 	 * Returns the frequency of the sine wave.
diff --git a/src/fx/DynamicMusic.cpp b/src/fx/DynamicMusic.cpp
index 2b0acc0..c682108 100644
--- a/src/fx/DynamicMusic.cpp
+++ b/src/fx/DynamicMusic.cpp
@@ -15,9 +15,6 @@
 ******************************************************************************/
 
 #include "fx/DynamicMusic.h"
-#include "generator/Silence.h"
-#include "fx/Fader.h"
-#include "fx/Limiter.h"
 
 #include <mutex>
 #include <condition_variable>
diff --git a/src/fx/ModulatorReader.cpp b/src/fx/ModulatorReader.cpp
index c17cee0..f7cdc1d 100644
--- a/src/fx/ModulatorReader.cpp
+++ b/src/fx/ModulatorReader.cpp
@@ -48,7 +48,7 @@ int ModulatorReader::getLength() const
 	int len2 = m_reader2->getLength();
 	if((len1 < 0) || (len2 < 0))
 		return -1;
-	return std::min(len1, len2);
+	return std::max(len1, len2);
 }
 
 int ModulatorReader::getPosition() const
diff --git a/src/sequence/SuperposeReader.cpp b/src/sequence/SuperposeReader.cpp
index 9206a7a..a65185a 100644
--- a/src/sequence/SuperposeReader.cpp
+++ b/src/sequence/SuperposeReader.cpp
@@ -48,7 +48,7 @@ int SuperposeReader::getLength() const
 	int len2 = m_reader2->getLength();
 	if((len1 < 0) || (len2 < 0))
 		return -1;
-	return std::min(len1, len2);
+	return std::max(len1, len2);
 }
 
 int SuperposeReader::getPosition() const
-- 
2.22.0

